src/elexmodel/models/BaseElectionModel.py

```diff
@@ -205,7 +193,9 @@ def get_unit_prediction_interval_bounds(self, reporting_units, nonreporting_unit
 
         # specifying self.features extracts the correct columns and makes sure they are in the correct
         # order. Necessary when fitting and predicting on the model.
-        train_data_features = train_data[self.features]
+        # the fixed effects in train_data will be a subset of the fixed effect of reporting_units since all
+        # units from one fixed effect category might be in the conformalization data.
+        train_data_features = self.featurizer.featurize_fitting_data(train_data)
```

Comment:
I think there's an issue with your center_features flag. Right now you have it that those means (which are computed once when the Featurizer is initiated) get subtracted from the features in every call of the your featurizing functions. (I don't see you setting the flag to false in any of the calls). Don't we just want to do this once? Like maybe when you compute the means for centuring in line 52, you should also subtract the means from all the units (reporting+nonreporting) then never do it again? Is that right?

Reply:
hmm, I'm not sure that is true. We pass in the dataframe that we want to featurize, copy it and then subtract the means. So we're never subtracting the means from `reporting_units` (we're subtracting it from a copy of reporting_units). So when we pass in `train_data` (the subset of `reporting_units`) those columns have never been touched.

Reply:
ah ok -- yea i think this is fine then